File: algorithms/sorting/insertion-sort/javascript/insertion-sort.js
'use strict';

function insertionSort(array) {
    var i;
    for (i = 1; i < array.length; i++) {
        var item = array[i];
        var indexHole = i;
        while (indexHole > 0 && array[indexHole - 1] > item) {
            array[indexHole] = array[--indexHole];
        }
        array[indexHole] = item;
    }
    return array;
}

File: algorithms/sorting/InsertionSort.java
package algorithms.sorting;

public class InsertionSort {
    public static void sort(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int item = array[i];
            int indexHole = i;
            while (indexHole > 0 && array[indexHole - 1] > item) {
                array[indexHole] = array[--indexHole];
            }
            array[indexHole] = item;
        }
    }
}
